.github/workflows/install-python-dependencies.yml

```diff
@@ -0,0 +1,28 @@
+name: Install and cache Python dependencies
+
+on:
+  workflow_call:
+
+jobs:
+  build:
+    name: Install and cache Python dependencies with pip
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+
+      - uses: actions/cache@v3
+        id: cache
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ runner.os }}-python-${{ env.pythonLocation }}-${{ hashFiles('pyproject.toml') }}-test
+
+      - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -e .[server,postgresql,tests,integrations,listeners]
```

Comment:
Isn't there a command to install with `--all-extras`? this might lead to unwanted side effects w.r.t. changes we make in the extras.